### PR TITLE
Harvest and index configured collections

### PR DESCRIPTION
**Pull request by bess**
Work toward #9

### CODE REVIEW

File: app/lib/research_data_harvester.rb [hunk 1]
```diff
@@ -15,4 +16,28 @@ def collections_to_index
     end
     collections
   end
+
+  ##
+  # For a given ResearchDataCollection, retrieve its metadata from DataSpace
+  # @param [ResearchDataCollection] research_data_collection
+  def harvest(research_data_collection)
+    server = 'https://dataspace-dev.princeton.edu/rest'
```
Comment (tpendragon) [inline app/lib/research_data_harvester.rb:24]: suggestion: Extract to instance method.

File: app/lib/research_data_harvester.rb [hunk 2]
```diff
@@ -15,4 +16,28 @@ def collections_to_index
...
+    collection_id = research_data_collection.collection_id
```
Comment (tpendragon) [inline app/lib/research_data_harvester.rb:25]: question: Are you expecting to use other properties of the collection here? If not, maybe just pass the collection ID?

File: app/lib/research_data_harvester.rb [hunk 3]
```diff
@@ -15,4 +16,28 @@ def collections_to_index
...
+    Indexer.new(resp.body).index
+  end
+
+  ##
+  # Convenience method to harvest and index all collections in the config file
+  # @example
+  #   ResearchDataHarvester.harvest
+  def self.harvest
+    r = ResearchDataHarvester.new
+    r.collections_to_index.each do |collection|
```
Comment (tpendragon) [inline app/lib/research_data_harvester.rb:38]: Suggestion: Standardize on either `collection` or `research_data_collection`

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -99,5 +99,7 @@
   #   Kernel.srand config.seed
 end
 
+# WebMock.enable_net_connect!
```
Comment (tpendragon) [inline spec/spec_helper.rb:102]: Question: Is this meant to be left here? If so, maybe a comment to say why?

**Review by tpendragon — CHANGES_REQUESTED** [review]
Just a few things. If you have strong feelings against any of these changes I don't think it'd be a problem to merge without them.

File: app/lib/research_data_harvester.rb (hunk 2 above)
Comment (bess) [reply app/lib/research_data_harvester.rb:25]: Yes, we will need the other properties of collection when we're indexing, I just haven't written that part yet.

File: spec/spec_helper.rb (hunk 4 above)
Comment (bess) [reply spec/spec_helper.rb:102]: Comment added.

File: app/lib/research_data_harvester.rb (hunk 3 above)
Comment (bess) [reply app/lib/research_data_harvester.rb:38]: I standardized on `collection`.

**Review by jrgriffiniii — APPROVED** [review]

File: app/lib/research_data_harvester.rb (hunk 1 above)
Comment (bess) [reply app/lib/research_data_harvester.rb]: Good idea! Done!

**Pull request merged**